source/common/http/conn_manager_utility.cc

```diff
@@ -206,11 +206,14 @@ Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequest
   }
 
   // Generate x-request-id for all edge requests, or if there is none.
-  if (config.generateRequestId() && (edge_request || !request_headers.RequestId())) {
+  if (config.generateRequestId()) {
     // TODO(PiotrSikora) PERF: Write UUID directly to the header map.
-    const std::string uuid = random.uuid();
-    ASSERT(!uuid.empty());
-    request_headers.insertRequestId().value(uuid);
+    if ((!config.edgeAcceptRequestId() && (edge_request || !request_headers.RequestId())) ||
+        (config.edgeAcceptRequestId() && !request_headers.RequestId())) {
```

Comment:
This expression looked redundant, as it mentions 2 variables twice, but I was lazy with the truth-tables and karnaugh-maps so I ran it through it into https://www.dcode.fr/boolean-expressions-calculator and found it equivalent to:
```
(! config.edgeAcceptRequestId() && edge_request) || ! request_headers.RequestId()
```
maybe it's easier to read that way too?

It's probably worth writing some unit tests covering the 8 cases for this?

Reply:
sure thing thanks 👍